model/task_lifecycle.go

```diff
@@ -293,21 +293,17 @@ func TryResetTask(ctx context.Context, settings *evergreen.Settings, taskId, use
 
 	maxExecution := evergreen.MaxTaskExecution
 
-	if evergreen.IsCommitQueueRequester(t.Requester) && evergreen.IsSystemFailedTaskStatus(t.Status) {
-		maxSystemFailedTaskRetries := settings.CommitQueue.MaxSystemFailedTaskRetries
-		if maxSystemFailedTaskRetries > 0 {
-			maxExecution = maxSystemFailedTaskRetries
-		}
-	}
-	// if we've reached the max number of executions for this task, mark it as finished and failed
-	if t.Execution >= maxExecution {
-		// restarting from the UI bypasses the restart cap
-		msg := fmt.Sprintf("task '%s' reached max execution %d: ", t.Id, maxExecution)
-		if origin == evergreen.UIPackage || origin == evergreen.RESTV2Package || origin == evergreen.GithubCheckRun {
-			grip.Debugln(msg, "allowing exception for", user)
-		} else if !t.IsFinished() {
+	// For system failures, we restart once for tasks on their first execution, if configured.
+	if evergreen.IsSystemFailedTaskStatus(t.Status) && t.Execution == 0 &&
```

Comment:
I think it would make more sense to have `evergreen.IsSystemFailedTaskStatus(t.Status) && t.Execution == 0 && settings.ServiceFlags.SystemFailedTaskRestartDisabled` be its own bool value to condition on at L304, rather than changing the maxExecution value to one.

I think this would be easier to understand since the desicion we're making here is a binary one (i.e. we are either restarting it once or we aren't)

Reply:
I thought about that, but I think that's actually not what this boolean means with the way it's implemented -- really it only restarts if the _first_ execution failed. If we just check the boolean at 304 without considering execution or adding some new state tracking, we'll always restart bc we won't know if we've auto-restarted once or not.

I'm not inclined to add state tracking for this since ideally we usually won't be in this degraded mode.

Reply:
But I think I think we should be setting maxExecution = 1 for all system failed tasks and not only for tasks where `t.Execution == 0` correct?

Reply:
Also I think the way we're checking system failure here is outdated, but hasn't mattered since https://github.com/evergreen-ci/evergreen/blob/main/model/task_lifecycle.go#L2223 does a check also

Reply:
I'm actually not certain we need this additional check but i think it doesn't hurt